packages/echo/src/client.ts

```diff
@@ -503,6 +511,14 @@ export class Echo {
     };
   }
 
+  private async shouldSkip(skip: Skip | undefined, payload: Record<string, unknown>): Promise<boolean> {
+    if (!skip) {
+      return false;
+    }
+
+    return skip(payload);
+  }
+
```

Comment:
The `payload` is freely usable in the skip function after hydration occurs. It's the `inputs` which are contextually available only for this step. We need to update naming to reflect this.

```suggestion
  private async shouldSkip(skip: Skip | undefined, inputs: Record<string, unknown>): Promise<boolean> {
    if (!skip) {
      return false;
    }

    return skip(inputs);
  }

```